pkg/aws/aws.go

```diff
@@ -145,77 +136,51 @@ func (c *SdkClient) AssumeRole(roleARN, region string) (*SdkClient, error) {
 }
 
 // ListRunningInstances lists all running or starting instances that belong to a cluster
-func (c *SdkClient) ListRunningInstances(infraID string) ([]*ec2.Instance, error) {
-	filters := []*ec2.Filter{
-		{
-			Name:   aws.String("tag:kubernetes.io/cluster/" + infraID),
-			Values: []*string{aws.String("owned")},
-		},
-		{
-			Name:   aws.String("instance-state-name"),
-			Values: []*string{aws.String("running"), aws.String("pending")},
-		},
-	}
-	return c.listInstancesWithFilter(filters)
-}
-
-// ListNonRunningInstances lists all non running instances that belong to a cluster
-func (c *SdkClient) ListNonRunningInstances(infraID string) ([]*ec2.Instance, error) {
-	filters := []*ec2.Filter{
-		{
-			Name:   aws.String("tag:kubernetes.io/cluster/" + infraID),
-			Values: []*string{aws.String("owned")},
-		},
-		{
-			Name: aws.String("instance-state-name"),
-			Values: []*string{
-				aws.String("stopped"),
-				aws.String("stopping"),
-				aws.String("terminated"),
-				aws.String("terminating"),
+func (c *SdkClient) ListRunningInstances(infraID string) ([]ec2v2types.Instance, error) {
+	output, err := c.Ec2Client.DescribeInstances(context.TODO(), &ec2v2.DescribeInstancesInput{
+		Filters: []ec2v2types.Filter{
+			{
+				Name:   aws.String("tag:kubernetes.io/cluster/" + infraID),
+				Values: []string{"owned"},
+			},
+			{
+				Name:   aws.String("instance-state-name"),
+				Values: []string{"running", "pending"},
 			},
 		},
+	})
+	if err != nil {
+		return nil, err
 	}
-	return c.listInstancesWithFilter(filters)
+	return output.Reservations[0].Instances, nil
```

Comment:
**Question:** can this result in a panic? We're not sure there's reservations to my understanding. Previous code iterated through reservations and appended instances.

Reply:
I was thinking of putting in paginator, but that would need more mocking on top - https://aws.github.io/aws-sdk-go-v2/docs/unit-testing/#mocking-paginators - so I think we can live without mocking those for now.